qiskit/providers/ibmq/jupyter/dashboard/watcher_monitor.py

```diff
@@ -12,84 +12,160 @@
+    def __init__(self, job: Job, watcher: 'IQXDashboard') -> None:
+        """Creates a JobMonitor.
+
+        Args:
+            job: the job
+            watcher: the dashboard
+        """
+        self.job = job
+        self._job_type = get_job_type(job)
+        self._watcher = watcher
+        self._prev_status_name = None
+        self._interval = 0.5
+        self._exception_count = 0
+
+    def _update(self, info: Tuple) -> None:
+        """Updates the watcher with the provided info"""
+        self._watcher.update_single_job(self._job_type, info)
+
+    @abstractmethod
+    def _set_job_queued(self) -> None:
+        """Perform updates particular to the QUEUED phase"""
+        pass
+
+    def start(self):
+        """Start the monitor"""
+        status = self.job.status()
+        while status.name not in ['DONE', 'CANCELLED', 'ERROR']:
+            time.sleep(self._interval)
+            try:
+                status = self.job.status()
+                self._exception_count = 0
+                if status.name == 'QUEUED':
+                    self._set_job_queued()
+                elif status.name != self._prev_status_name:
+
+                    msg = status.name
+                    info = (self.job.job_id(), msg, 0, status.value)
+
+                    # Update the job on the dashboard
+                    self._update(info)
+
+                    self._interval = 2
+                    self._prev_status_name = status.name
+
+            # pylint: disable=broad-except
+            except Exception:
+                self._exception_count += 1
+                if self._exception_count == 5:
+                    info = (self.job.job_id(), 'NA', 0, "Could not query job.")
+                    # Update the job on the dashboard
+                    self._update(info)
+                    sys.exit()
+
+
+class RuntimeJobMonitor(JobMonitor):
+    """Monitors a runtime job for updates to display to dashboard"""
+
+    def __init__(self, job: RuntimeJob, watcher: 'IQXDashboard') -> None:
+        """Initialize a runtime job
+
+        Args:
+            job: the job
+            watcher: the dashboard
+        """
+        super().__init__(job, watcher)
+
+    def _set_job_queued(self) -> None:
+        """Perform queued updates"""
+        info = (self.job.job_id(), self.job.status().name)
+        # Update the job on the dashboard
+        self._update(info)
+
+
+class CircuitJobMonitor(JobMonitor):
+    """Monitors a circuit job for updates to display to dashboard"""
+
+    def __init__(self, job: IBMQJob, watcher: 'IQXDashboard') -> None:
+        """Initialize a circuit job
+
+        Args:
+            job: the job
+            watcher: the dashboard
+        """
+        super().__init__(job, watcher)
+        self._prev_queue_pos = None
+        self._prev_est_time = ''
+
+    def _set_job_queued(self) -> None:
+        """Perform queued updates"""
+        status = self.job.status()
```

Comment:
If you are re-querying job status here, it may no longer be `QUEUED`, in which case you shouldn't display queue position / est. start time.

Reply:
Fixed! This function now checks if the status is still `queued`.